pkg/app/run.go

```diff
@@ -174,7 +175,7 @@ func Run(opts RunOpts) error {
 		return fmt.Errorf("error constructing Fastly realtime stats client: %w", err)
 	}
 
-	if opts.Versioners.CLI != nil && name != "update" && !version.IsPreRelease(revision.AppVersion) {
+	if opts.Versioners.CLI != nil && name != "update" && !version.IsPreRelease(revision.AppVersion) && !cmd.IsCompletion(opts.Args) && !cmd.IsCompletionScript(opts.Args) {
```

Comment:
```suggestion
	// NOTE: We don't want to trigger a config check when the user is making an 
	// autocomplete request because this can add additional latency to the user's 
	// shell loading completely.
	if opts.Versioners.CLI != nil && name != "update" && !version.IsPreRelease(revision.AppVersion) && !cmd.IsCompletion(opts.Args) && !cmd.IsCompletionScript(opts.Args) {
```